src/all/yellownote/src/eu/kanade/tachiyomi/extension/all/yellownote/YellowNotePreferences.kt

```diff
@@ -13,17 +15,22 @@ object YellowNotePreferences {
     private const val PS_KEY_DOMAIN = "$PS_KEY_ROOT::DOMAIN"
     private const val PS_KEY_DOMAIN_DEFAULT = "$PS_KEY_DOMAIN::DEFAULT"
     private const val PS_KEY_DOMAIN_OVERRIDE = "$PS_KEY_DOMAIN::OVERRIDE"
+    private const val PS_KEY_LANGUAGE = "$PS_KEY_ROOT::LANGUAGE"
 
     private const val DEFAULT_DOMAIN = "https://xchina.co"
-
-    private fun SharedPreferences.getStringOrDefaultIfBlank(key: String, defValue: String): String {
-        val value = getString(key, defValue)!!
+    private fun SharedPreferences.getStringOrDefault(key: String, defValue: String): String {
+        val value = getString(key, defValue) ?: defValue
         return value.ifBlank { defValue }
     }
 
+    private fun SharedPreferences.getStringOrDefault(key: String, defValueSupplier: () -> String): String {
+        val value = getString(key, "") ?: defValueSupplier.invoke()
+        return value.ifBlank { defValueSupplier.invoke() }
+    }
```

Comment:
Remove these helpers. `getString(key, NON_NULL_VALUE)` is guaranteed to not return a non-null value and you can chain `!!.ifBlank { }` if needed.